addons/payment_stripe/controllers/main.py

```diff
@@ -55,6 +55,15 @@ def stripe_s2s_create_setup_intent(self, acquirer_id, **kwargs):
     def stripe_s2s_process_payment_intent(self, **post):
         return request.env['payment.transaction'].sudo().form_feedback(post, 'stripe')
 
+    @http.route('/payment/stripe/s2s/process_payment_error', type='json', auth='public', csrf=False)
+    def stripe_s2s_process_payment_error(self, **post):
+        transaction = request.env['payment.transaction'].sudo().search([('reference', '=', post['reference']),
+                                                                        ('stripe_payment_intent', '=', post['stripe_payment_intent']),
+                                                                        ('stripe_payment_intent_secret', '=', post['stripe_payment_intent_secret'])])
+        transaction.state = 'error'
+        transaction.state_message = post['error']
```

Comment:
Update the record in one call.
```suggestion
        transaction.write({'state: 'error', 'state_message': post['error']})
```